cloud/hack/test-integration.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(unset CDPATH && cd $(dirname "${BASH_SOURCE[0]}")/../../ && pwd)
+export KUBE_CONFIG=""
+export KUBE_APISERVER_URL="http://localhost:8080"
+
+runTests() {
+  cleanup
+
+  logStatus "Applying device crd yamls"
+  kubectl create -f "${SCRIPT_ROOT}/build/crds/devices/devices_v1alpha1_devicemodel.yaml"
+  kubectl create -f "${SCRIPT_ROOT}/build/crds/devices/devices_v1alpha1_device.yaml"
+
+  logStatus  "Running integration test cases"
+  export KUBE_RACE="-race"
+  make -C "${SCRIPT_ROOT}/cloud" test \
+      WHAT="${WHAT:-$(findIntegrationTestDirs | paste -sd' ' -)}" \
+      GOFLAGS="${GOFLAGS:-}" \
+      KUBE_TEST_ARGS="--alsologtostderr=true ${KUBE_TEST_ARGS:-} " \
+      KUBE_RACE="" \
+
+  cleanup
+}
+
+findIntegrationTestDirs() {
+  (
+    cd "${SCRIPT_ROOT}/cloud"
+    find test/integration/ -name '*_test.go' -print0 \
+      | xargs -0n1 dirname | LC_ALL=C sort -u
+  )
+}
+
+cleanup() {
+  logStatus "Cleaning up"
+
+  local num_devices=$(kubectl get device | wc -l) && rc=$?
```

Comment:
Maybe use a random namespace for testing and tear it down afterwards?
Manually deleting the devices and devicemodels look inflexible.

Reply:
Yes, ideally we should be doing this programmatically, however, when I tried using the 
k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset libraries, I ran into dependency issues which can't be resolved until we upgrade to new Kubernetes version.

So I have kept the setup and teardown as part of the scripts itself.

Reply:
we just verify the validation of device CRD here, how about run and set --dryrun flag, so no need do the cleanup

Reply:
The CRD definitions need to be installed in the cluster in order to create the device model and instances and validate.